addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/SingleWebDriverProvider.java

```diff
@@ -64,6 +64,22 @@
      */
     WebDriver getWebDriver(int requesterId, String proxyAddress, int proxyPort);
 
+    /**
+     * Gets a {@code WebDriver} to the provided browser for the given requester, proxying through
+     * the given address and port.
+     *
+     * @param requesterId the ID of the (ZAP) component that's requesting the {@code WebDriver}.
+     * @param proxyAddress the address of the proxy.
+     * @param proxyPort the port of the proxy.
+     * @param enableExtensions if true then optional browser extensions are enabled
+     * @return the {@code WebDriver} to the provided browser, proxying through the given address and
+     *     port.
+     * @throws IllegalArgumentException if {@code proxyAddress} is {@code null} or empty, or if
+     *     {@code proxyPort} is not a valid port number (between 1 and 65535).
+     */
+    WebDriver getWebDriver(
```

Comment:
This should not be needed (hopefully).

Reply:
In theory someone else could be using it?

Reply:
This interface was only added to support JxBrowser.

Reply:
Its used by the Quick Start add-on, and I like the idea of giving other add-ons access to this level of control...

Reply:
That's unrelated to the interface though, which is no longer used by any add-on (at least in this repo where it was initially used).